CONTRIBUTING.md

```diff
@@ -28,65 +28,118 @@ Here are questions you can answer before you file a bug to make sure you're not
+Title tags  are generally one word or acronym enclosed in square brackets.
+Limiting to one or two tags is ideal to keep titles short.
+Some examples of common tags are:
+
+- `[NFC]` - No Functional Change
+  - `[RFC]` - Request For Comments (often used for drafts to get feedback)
+  - `[Doc]` - Documentation change
+  - `[SPIRV]` - Changes related to SPIR-V
+  - `[HLSL2021]` - Changes related to HLSL 2021 features
+  - Other tags in use: `[Linux]`, `[mac]`, `[Win]`, `[PIX]`, etc...
+
+Tags aren't formalized or any specific limited set. If you're unsure of
+  a reasonable tag to use, just don't use any. If you want to invent a new
+  tag, go for it! These are to help categorize changes at a glance.
+
+#### Descriptions
+
+The PR description should include a more detailed description of the change,
+an explanation for the motivation of the change, and links to any relevant Issues.
+This does not need to be a dissertation, but should leave
+breadcrumbs for the next person debugging your code (who might be you).
+
+Using the words `Fixes`, `Fixed`, `Closes`, `Closed`, or `Close` followed by
+  `#<issuenumber>`, will auto close an issue after the PR is merged.
+
+#### Release Notes
+
+Significant changes may require release notes that highlight important
+compiler behavior changes for each named release.
+These include changes that are:
+
+- Visible to the users
+- Significant changes to compiler behavior
+  - New features: Language, Hardware support, compiler options
+  - Important bug fixes
+  - Changes in default behavior
+
+When such a change is made, the release note should be included as part of that change.
+This is done in the docs/ReleaseNotes.md file.
+
+If the change is meant for a named release,
+it should be added to that named release's section of the release notes file.
+As the change is merged to the appropriate release branches,
+the release notes will come along with it.
+
+If a change is meant for the next upcoming release,
+it should be added to the "Upcoming Release" section.
+When the next upcoming release is named,
+the title will be updated and the release note will be included in the appropriate release.
+
+When writing release note list entries:
+
+- Keep the description to a single sentence.
+- Links to PRs are not recommended.
+- Markdown links to bugs are fine if the issue is too complicated to completely explain in a single sentence.
+- Remember to update release notes as the nature of the change alters or is removed.
+
+### Testing Pull Requests
+
+For a pull request to be merged, it will have to pass the automated set of regression tests run for each.
+Additional regression testing may be required for some changes depending on the area of the changes.
+The commiter is expected to recognize the need for and perform any additional testing prior to merging.
+
+In addition to existing tests, bug fixes and new features require additional testing be included in the pull requests that implement them.
```

Comment:
I would put this at the top of the section on testing since it is the most important thing. Maybe even add something explicit like:

> All PRs that contribute functional or behavioral changes to the compiler must have additional tests that verify the new code behaves correctly. Changes that do not change behavior may be required to add testing if the change impacts areas with poor test coverage.

Reply:
Moved. I tried to meld the new suggestions with the existing text.